src/commands/publish.rs

```diff
@@ -62,11 +78,25 @@ pub fn publish(
         // Next, upload and deploy the worker with the updated asset_manifest
         upload::script(&upload_client, &target, Some(asset_manifest))?;
 
-        deploy::worker(&user, &deploy_config)?;
+        match deploy::worker(&user, &deploy_config) {
+            Ok((result_msg, urls)) => {
+                if is_json_out {
+                    jsonout.success = true;
+                    jsonout.name = target.name.clone();
+                    jsonout.urls = urls;
```

Comment:
construct your PublishResult struct here (and below) rather than instantiating it outside the if/else